sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -910,6 +957,25 @@ def predict(self, X):
         # shape (n_samples, 1) to (n_samples,)
         return self.loss_.inverse_link_function(self._raw_predict(X).ravel())
 
+    def staged_predict(self, X):
+        """Predict regression target at each stage for X.
+
+        This method allows monitoring (i.e. determine error on testing set)
+        after each stage.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            The input samples.
+
+        Returns
+        -------
+        y : generator of ndarray of shape (n_samples,)
+            The predicted value of the input samples.
+        """
+        for raw_predictions in self._staged_raw_predict(X):
+            yield raw_predictions.ravel()
```

Comment:
I guess you could just use `yield from self._staged_raw_predict(X)` though the current form is totally fine